proposal.md

```diff
@@ -1 +1,100 @@
-**Include your feature proposal here**
+This proposal is a mostly a list of ideas thrown together, there is much to discuss. Basically, the goal is to add support for multiple workspaces, add flexibility on authentication and make it easier to implement at scale.
+
+## Terms
+
+- Chat Users: refers to users speaking with a bot (which may or may not be authenticated)
+- Collaborators: Authenticated users on the studio
+
+## Summary of proposed changes
+
+- Adds support for multiple workspaces (a workspace has custom user roles and hosts multiple bots)
+- Remove collaborators from "workspaces.json" and move them to the database
+- Collaborators accounts are global (may be in one or more workspaces)
+- Extract "Auth Strategies" from botpress config into its own file (there could be multiple basic auth or saml auth config)
+- Users would be authenticated by a UUID, so there could be multiple different tables for different auth strategies
+- Add a new skill, "Authentication Gate", which would let the user through if he's authenticated, or force him to authenticate.
+- Add an "Auth Strategy" property optionally to each bots. It will be used when faced with an Authentication Gate
```

Comment:
Auth Strategy is set on the workspace (not individual bots), and end users are added/removed for the whole workspace too

Reply:
I was thinking that setting the strategy on individual bots allows for a bit more flexibility (ex: one public facing bot, and one for internal staff) in the same workspace.